### DISCUSSION TITLE
Progress on The New Forge Notebooks

### DISCUSSION THREAD

**Discussion by ffxvs** [ffxvs/sd-webui-complete-setup#11] [Announcements]
First of all, thank you for using my notebooks. In the last few weeks, I’ve been reworking the entire code in this project. The goal is to make installing and delivering updates easier. But it turns out, it's taking longer than I expected. Sorry about that. This update not only affects Forge notebooks, but also Auto1111 notebooks.

I know you are looking forward to using the new Forge. The good news is that the new Forge supports Flux, the most advanced model for image generation. If you don’t know about it, I recommend checking it out on r/StableDiffusion. 

The bad news is that the new Forge uses Gradio 4 (a framework for building user interfaces) which breaks almost all extensions. Beside, many extensions haven’t been updated in months. I haven’t tested all the extensions, but here are some that I’ve tried :

**Broken extensions :**
- [Aspect Ratio Helper](https://github.com/thomasasfk/sd-webui-aspect-ratio-helper) (partially working)
- [AnimateDiff](https://github.com/continue-revolution/sd-webui-animatediff)
- [BMAB](https://github.com/portu-sim/sd-webui-bmab)
- [Canvas Zoom](https://github.com/richrobber2/canvas-zoom) (broken, but already covered by Gradio 4)
- [Cleaner](https://github.com/novitalabs/sd-webui-cleaner)
- [State](https://github.com/ilian6806/stable-diffusion-webui-state) (partially working)
- [Inpaint Anything](https://github.com/Uminosachi/sd-webui-inpaint-anything)
- [Latent Couple](https://github.com/aria1th/stable-diffusion-webui-two-shot)
- [Mini Paint](https://github.com/0Tick/a1111-mini-paint) (working, but send buttons doesn't work)
- [Regional Prompter](https://github.com/hako-mikan/sd-webui-regional-prompter)
- Generally, all extensions that using Gradio Canvas to manipulate image.

Some extensions like Adetailer and TagAutoComplete are still working. So, if you are using the above extension, I suggest you look for an alternative or go back to the original Auto1111's Webui.

## Current Progress

**I dare to say that currently the new Forge's notebooks are almost 100% complete** and working fine for Paperspace and RunPod. But I can't release them yet because the resources used are also shared in Auto1111's notebooks. So I have to work on that one too. But it won't take long.

### But you can try the notebooks in the dev branch  

- For RunPod user, use the [SD Web UI Forge](https://runpod.io/console/deploy?template=kwef1wl832&ref=synjcfeg) template, then change the Container Image to 
```
ffxvs/sd-webui-containers:forge-test
```

- For Paperspace user, create a new notebook in the same project. Then fill in the field :
  - Container's name
  ```
  ffxvs/sd-webui-containers:forge-test
  ```

  - Container's command
  ```
  bash /paperspace-start.sh
  ```

This dev branch is temporary, so please use it only until the new update arrive. When your current notebook is not working, that means a new update has arrived on the latest channel. So, please create a fresh installation.

**Comment by ffxvs** [comment]
Installing extensions and downloading models will be easier now.

![extensions_selector](https://github.com/user-attachments/assets/496eea6b-2ee4-461f-af5e-4f7ccaf28682)

<br>

![models_selector](https://github.com/user-attachments/assets/0224cc13-2edd-441c-8712-ada1765a8dc4)

I am open to feedback

**Comment by Normiegetout** [comment]
Thank you for your hard work!

**Comment by ffxvs** [comment]
- Auto1111's notebook is completed.
- Temporary storage settings (Paperspace) now have an UI.

Everything is pretty much done. Controlnet and Install from URLs still use the old way. I will change them later after the update.

**Comment by ffxvs** [comment]
All the latest notebooks are in this container now. Will merge it into the main branch soon.

```
ffxvs/sd-webui-containers:forge-test
```

```
ffxvs/sd-webui-containers:auto1111-test
```
